# PDA Project Dec 2018 
by Colm Doherty


## 1. Researching an existing dataset

dataset source : Met Eireann https://www.met.ie
monthly rainfall at Dun Laoghaire weather station, 1997-2018
        

In [6]:
# Import pandas etc.
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
! ls


Data_Licence.pdf           MET9223.csv
Data_Licence.txt           PDA Project Dec 2018.ipynb
Hello.py                   PDA project Dec.pdf
Indicator.txt              README.md
LICENSE


In [8]:
# Load the dataset
import csv as df
df = pd.read_csv("MET9223.csv", skiprows=13) 

In [9]:
df

year  month   rain   gdf  rd  wd
0    1997      8   91.8  17.4  18  16
1    1997      9   11.5   2.9  12   4
2    1997     10   70.7  20.7  14  10
3    1997     11  187.4  54.6  22  17
4    1997     12   84.4  21.7  14  12
5    1998      1  109.4  16.5  19  15
6    1998      2    8.4   2.9   8   4
7    1998      3   61.3  10.3  15  11
8    1998      4  136.5    39  26  16
9    1998      5   27.7  12.5  11   5
10   1998      6  125.1  29.1        
11   1998      7   34.2              
12   1998      8   33.8  13.2   9   7
13   1998      9   89.4  21.6  18  17
14   1998     10   73.5     8  23  15
15   1998     11   87.1    23  19  14
16   1998     12   74.8  28.7  23  17
17   1999      1   61.8   9.3  22  12
18   1999      2   22.4   3.9  15   6
19   1999      3   19.9   4.8  13   5
20   1999      4   67.7  15.6  19  14
21   1999      5   48.1  14.5  14  10
22   1999      6   58.2  22.5  15  11
23   1999      7   19.7              
24   1999      8  119.6    41  15   9
25   1999      9  139.8  27.8  20  13
26   1999     10   42.9   9.3  13   7
27   1999     11   41.9  11.1  16   6
28   1999     12   61.0  12.3  20  12
29   2000      1   26.1   9.5  14   6
..    ...    ...    ...   ...  ..  ..
223  2016      3   29.7   5.7  14   8
224  2016      4   55.6  29.7  13   9
225  2016      5   38.7   7.3  13   9
226  2016      6   57.5   8.5  16  14
227  2016      7   28.0     4  17  11
228  2016      8   55.5  26.6  15   8
229  2016      9   68.6  20.4  16  14
230  2016     10   38.6  15.8   8   4
231  2016     11   31.0   8.4  13   7
232  2016     12   55.0  13.2  15  11
233  2017      1   19.3  10.5  13   5
234  2017      2   66.5  12.7  16  12
235  2017      3   57.3  14.9  16  12
236  2017      4    9.9     4   7   3
237  2017      5   61.5    32  16   9
238  2017      6   84.3    18  18  12
239  2017      7   38.4  11.1  12   7
240  2017      8   53.1  17.5  15   9
241  2017      9   91.5    19  23  13
242  2017     10   55.1  14.9  15  10
243  2017     11   72.3  29.8  15  10
244  2017     12   57.9  16.2  15  10
245  2018      1   91.8    18  21  17
246  2018      2   29.6   7.4  15  11
247  2018      3  113.1    36  22  18
248  2018      4   78.4  12.1  21  16
249  2018      5   20.1     6  10   5
250  2018      6    1.2     1   2   1
251  2018      7   43.5  12.3   9   7
252  2018      8   27.8  11.4  14  10

[253 rows x 6 columns]

In [10]:
df.head()


year  month   rain   gdf  rd  wd
0  1997      8   91.8  17.4  18  16
1  1997      9   11.5   2.9  12   4
2  1997     10   70.7  20.7  14  10
3  1997     11  187.4  54.6  22  17
4  1997     12   84.4  21.7  14  12

In [16]:
df.tail()


year  month  rain   gdf  rd  wd
248  2018      4  78.4  12.1  21  16
249  2018      5  20.1     6  10   5
250  2018      6   1.2     1   2   1
251  2018      7  43.5  12.3   9   7
252  2018      8  27.8  11.4  14  10

In [24]:
# what does the data distribution look like?
df.describe()

year       month        rain
count   253.000000  253.000000  253.000000
mean   2007.624506    6.505929   66.385375
std       6.110532    3.453340   40.455267
min    1997.000000    1.000000    1.200000
25%    2002.000000    4.000000   36.600000
50%    2008.000000    7.000000   57.500000
75%    2013.000000    9.000000   88.300000
max    2018.000000   12.000000  207.000000

In [25]:
# ok, lets ignore the Year & Month columns
df1 = df[['rain','gdf','rd','wd']]
df1.describe()

rain
count  253.000000
mean    66.385375
std     40.455267
min      1.200000
25%     36.600000
50%     57.500000
75%     88.300000
max    207.000000

notice that only the 'rain' column has a full count of 253 observations. All other variables are missing some data.